indico/web/flask/session_test.py

```diff
@@ -0,0 +1,58 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2023 CERN
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see the
+# LICENSE file for more details.
+
+from datetime import datetime, timedelta, timezone
+from unittest.mock import Mock
+
+import pytest
+from flask import Response
+
+from indico.web.flask.session import IndicoSession, IndicoSessionInterface
+
+
+def test_session_hard_expiry_dt_without_tz():
+    """Test that passing an unaware datetime raises an error."""
+    session = IndicoSession()
+
+    with pytest.raises(ValueError):
+        session.hard_expiry = datetime(2021, 1, 1, 0, 0, 0)
+
+
+def test_get_storage_lifetime_with_hard_expiry(app, mocker):
+    """Test that the storage lifetime is the hard expiry."""
+    dt_mock = Mock(spec_set=datetime)
+    dt_now = dt_mock.now.return_value = datetime.now(timezone.utc)
+    mocker.patch('indico.web.flask.session.datetime', dt_mock)
```

Comment:
I think you may be able to avoid manually mocking datetime altgeother and instead use our `freeze_time` pytest fixture.